test/lit/passes/type-ssa-exact.wast

```diff
@@ -0,0 +1,1690 @@
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.new $struct-new-exact
+  ;; CHECK-NEXT:    (local.get $used)
+  ;; CHECK-NEXT:    (ref.null none)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.new $used-in-struct-new
+  ;; CHECK-NEXT:    (ref.null none)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $struct-new (param $used (ref (exact $used-in-struct-new)))
+    (drop
+      ;; The outer new can still be optimized.
+      (struct.new $struct-new-exact
+        (local.get $used)
+        (ref.null none)
+      )
+    )
+    (drop
+      (struct.new $used-in-struct-new
+        (ref.null none)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $struct-new-ok (type $104) (param $used (ref (exact $used-in-struct-new-ok)))
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.new $struct-new-inexact
+  ;; CHECK-NEXT:    (local.get $used)
+  ;; CHECK-NEXT:    (ref.null none)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.new $used-in-struct-new-ok_21
+  ;; CHECK-NEXT:    (ref.null none)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $struct-new-ok (param $used (ref (exact $used-in-struct-new-ok)))
+    (drop
+      (struct.new $struct-new-inexact
+        (local.get $used)
+        (ref.null none)
+      )
+    )
+    (drop
+      (struct.new $used-in-struct-new-ok
+        (ref.null none)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $struct-set (type $105) (param $used (ref (exact $used-in-struct-set))) (param $ref (ref $struct-set-exact))
+  ;; CHECK-NEXT:  (struct.set $struct-set-exact 0
+  ;; CHECK-NEXT:   (local.get $ref)
+  ;; CHECK-NEXT:   (local.get $used)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.new $used-in-struct-set
+  ;; CHECK-NEXT:    (ref.null none)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $struct-set (param $used (ref (exact $used-in-struct-set))) (param $ref (ref $struct-set-exact))
+    (struct.set $struct-set-exact 0
+      (local.get $ref)
+      (local.get $used)
+    )
+    (drop
+      (struct.new $used-in-struct-set
+        (ref.null none)
+      )
+    )
+  )
+
+  ;; CHECK:      (func $struct-set-ok (type $106) (param $used (ref (exact $used-in-struct-set-ok))) (param $ref (ref $struct-set-inexact))
+  ;; CHECK-NEXT:  (struct.set $struct-set-inexact 0
+  ;; CHECK-NEXT:   (local.get $ref)
+  ;; CHECK-NEXT:   (local.get $used)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (struct.new $used-in-struct-set-ok_22
+  ;; CHECK-NEXT:    (ref.null none)
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $struct-set-ok (param $used (ref (exact $used-in-struct-set-ok))) (param $ref (ref $struct-set-inexact))
+    (struct.set $struct-set-inexact 0
+      ;; We _can_ optimize this struct.new, but that's also incidental.
```

Comment:
What does this refer to? It is on a `local.get`

Reply:
Ah, this is stale. The `local.get` used to be an allocation, but that doesn't validate yet because we don't type allocations as exact yet.